### PR TITLE
docs: fix PHPDoc type mixed

### PR DESCRIPTION
**Pull request by kenjis**
~~Needs #6722~~

**Description**
See #6310

- fix PHPDoc types
- ~~remove unneeded empty check~~

**Checklist:**
- [x] Securely signed commits
- [] Component(s) with PHPDoc blocks, only if necessary or adds value
- [] Unit testing, with >80% coverage
- [] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: system/BaseModel.php [hunk 1]
```diff
@@ -416,7 +416,7 @@ abstract protected function doDelete($id = null, bool $purge = false);
      * through soft deletes (deleted = 1)
      * This methods works only with dbCalls
      *
-     * @return bool|mixed
+     * @return bool|string
```
Comment (MGatner) [inline system/BaseModel.php:419]: What the heck is a string return here??

File: system/Common.php [hunk 2]
```diff
@@ -414,10 +414,11 @@ function env(string $key, $default = null)
      * If $data is an array, then it loops over it, escaping each
      * 'value' of the key/value pairs.
      *
-     * Valid context values: html, js, css, url, attr, raw
-     *
      * @param array|string $data
-     * @param string       $encoding
+     * @param string       $context  Valid context values: html, js, css, url, attr, raw
```
Comment (MGatner) [inline system/Common.php:418]: Let's make these literals, at least for PHPStan:
```suggestion
     * @param 'html'|'js'|'css'|'url'|'attr'|'raw' $context
```

**Review by MGatner — CHANGES_REQUESTED** [review]

File: system/BaseModel.php (hunk 1 above)
Comment (kenjis) [reply system/BaseModel.php]: When in test mode.

File: system/Common.php [hunk 3]
```diff
@@ -437,7 +438,7 @@ function esc($data, string $context = 'html', ?string $encoding = null)
             // Provide a way to NOT escape data since
             // this could be called automatically by
             // the View library.
-            if (empty($context) || $context === 'raw') {
+            if ($context === 'raw') {
```
Comment (samsonasik) [inline system/Common.php:441]: please revert, '0' is empty, ref https://3v4l.org/mgSdd

this PR should only change PHPDoc, change code may cause regression.

Comment (samsonasik) [reply system/Common.php:441]: If it needs to be applied, I think it should go to 4.3 as it changed behaviour.

Comment (kenjis) [reply system/Common.php:441]: @samsonasik Technically you are correct.

But I cannot imagine users who write code like `esc($string, '0')`.
I see no problem with this change in develop.

Comment (samsonasik) [reply system/Common.php:441]: Ok then 👍

Comment (kenjis) [reply system/Common.php:441]: @samsonasik I think accepting `esc($string, '0')` is a bug.
I'll fix it in another PR.
Thank you for commenting!

Comment (kenjis) [reply system/Common.php:441]: I sent #6722

**Review by MGatner — APPROVED** [review]

**Pull request merged**